cpplite/cpplite.debugger/test/unit/src/org/netbeans/modules/cpplite/debugger/ProcessParametersTest.java

```diff
@@ -0,0 +1,192 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.netbeans.modules.cpplite.debugger;
+
+import java.io.File;
+import java.io.OutputStream;
+import java.io.OutputStreamWriter;
+import java.io.Writer;
+import java.nio.file.Files;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import junit.framework.Test;
+import org.netbeans.api.extexecution.base.ExplicitProcessParameters;
+import org.netbeans.junit.NbModuleSuite;
+import org.openide.filesystems.FileObject;
+import org.openide.filesystems.FileUtil;
+
+/**
+ * Tests application parameters set through {@link ExplicitProcessParameters}.
+ *
+ * @author Martin Entlicher
+ */
+public class ProcessParametersTest extends AbstractDebugTest {
+
+    private static final String CPP_APP =
+            "#include <unistd.h>\n" +
+            "#include <stdio.h>\n" +
+            "#include <limits.h>\n" +
+            "\n" +
+            "int main(int argc, char **argv, char **envp) {\n" +
+            "   char buffer[PATH_MAX];\n" +
+            "   if (getcwd(buffer, sizeof(buffer)) != NULL) {\n" +
+            "       printf(\"CWD:%s\\n\", buffer);\n" +
+            "   } else {\n" +
+            "       perror(\"getcwd() error\");\n" +
+            "       return 1;\n" +
+            "   }\n" +
+            "   while (--argc) {\n" +
+            "      printf(\"ARG:%s\\n\", *(++argv));\n" +
```

Comment:
Printing arguments from last to first one!

Reply:
That's not true. `*(++argv)` provide results in the correct order.